servlet/src/main/java/io/undertow/servlet/handlers/security/ServletFormAuthenticationMechanism.java

```diff
@@ -195,27 +215,7 @@ protected void storeInitialLocation(final HttpServerExchange exchange, byte[] by
         if(!saveOriginalRequest) {
             return;
         }
-        final ServletRequestContext servletRequestContext = exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
-        final ServletContextImpl servletContextImpl =  servletRequestContext.getCurrentServletContext();
-        HttpSessionImpl httpSession = servletContextImpl.getSession(exchange, false);
-        boolean newSession = false;
-        if (httpSession == null) {
-            httpSession = servletContextImpl.getSession(exchange, true);
-            newSession = true;
-        }
-        Session session;
-        if (System.getSecurityManager() == null) {
-            session = httpSession.getSession();
-        } else {
-            session = AccessController.doPrivileged(new HttpSessionImpl.UnwrapSessionAction(httpSession));
-        }
-        if (newSession) {
-            int originalMaxInactiveInterval = session.getMaxInactiveInterval();
-            if (originalMaxInactiveInterval > authenticationSessionTimeout) {
-                session.setAttribute(ORIGINAL_SESSION_TIMEOUT, session.getMaxInactiveInterval());
-                session.setMaxInactiveInterval(authenticationSessionTimeout);
-            }
-        }
+        Session session = getAndInitializeSession(exchange, true);
```

Comment:
this should check for null, since its possible to get null session in some cases?

Reply:
https://github.com/undertow-io/undertow/blob/master/servlet/src/main/java/io/undertow/servlet/spec/ServletContextImpl.java#L973

Reply:
I think it is different, because the getAndInitializeSession returns null only if the second parameter, create is false? Unless I am missing something. I am merging as is, if I"m wrong, let's discuss tihs further and submit another fix @baranowb